File: prototypes/Simon/simon.py
import random,os,time
cp=[]
user=[]
score=1
sc=1

while True:
    cp.append(random.randint(0,10))
    for i in range(len(cp)):
        sc=0
        print(cp[i],end=" ")
    if sc==0:
        os.system('cls')
    for i in range(score):
        user_guess=int(input(f"\nenter {i+1} no: "))
        user.append(user_guess)
    # print("user array:",user)
    match=True
    for i in range(len(cp)):
        if cp[i]!=user[i]:
            match=False
            break
    if match:
        print("correctly guessed:")
        score+=1
    else:
        print(f"game over, your score :{score-1}")
        break
    user=[]
    

File: prototypes/Simon/codes/simon.rb
cp=[]
user=[]
score=1

loop do
    cp<<rand(0...10)
    print "in memory #{cp}\n"
    score.times do |i|
        p "\n enter a no:"
        user_guess=gets.chomp.to_i
        user<<user_guess
    end

    match=true
    cp.each_with_index do |num,i|       # enumerator in python
        if num!=user[i]
            match=false
            break
        end

    end
    if match
        p "correctly guessed"
        score+=1
    else
        print "game over score:#{score-1}"
        break
    end
    
    user=[]
end
